### PR TITLE
Add makefile target to generate in docker

### PR DESCRIPTION
**Pull request by carlosedp**
This PR improve the makefile:

- Clean mibs directory before re-download
- Add parameter for curl to follow redirects
- Improve generator makefile and readme to have a direct target

Signed-off-by: Carlos de Paula <me@carlosedp.com>

### CODE REVIEW

File: generator/README.md [hunk 1]
```diff
@@ -37,12 +37,9 @@ If you would like to run the generator in docker to generate your `snmp.yml` con
...
 
 This example will generate the example `snmp.yml` which is included in the top level of the snmp_exporter repo:
+
 ```sh
-make mibs
-docker build -t snmp-generator .
-docker run -ti \
-  -v "${PWD}:/opt/" \
-  snmp-generator generate
+docker-generate
```
Comment (SuperQ) [inline generator/README.md:42]: Shouldn't this be `make docker-generate`?

File: generator/Makefile [hunk 2]
```diff
@@ -81,7 +85,7 @@ docker-tag-latest:
 	docker tag "$(DOCKER_REPO)/$(DOCKER_IMAGE_NAME):$(DOCKER_IMAGE_TAG)" "$(DOCKER_REPO)/$(DOCKER_IMAGE_NAME):latest"
 
 .PHONY: mibs
-mibs: mib-dir \
+mibs: clean mib-dir \
```
Comment (SuperQ) [inline generator/Makefile:88]: Hmm, I'm not sure I would want to always `make clean` on every call to `make mibs`.

File: generator/README.md (hunk 1 above)
Comment (carlosedp) [reply generator/README.md]: My bad, typo :)
Fixed.

File: generator/Makefile (hunk 2 above)
Comment (carlosedp) [reply generator/Makefile:88]: The problem is that if you have an "old" mibs dir already in, running `make mibs` does nothing.

Comment (SuperQ) [reply generator/Makefile:88]: Yes, that's intentional. It's designed to be additive-only. Calling `make clean` should be something you do intentionally.

Comment (carlosedp) [reply generator/Makefile]: Ok, removed it.

File: snmp.yml [hunk 3]
```diff
@@ -4636,7 +4636,6 @@ cisco_wlc:
     indexes:
     - labelname: bsnAPDot3MacAddress
       type: PhysAddress48
-      fixed_size: 6
```
Comment (SuperQ) [inline snmp.yml:4639]: Strange, why is this being removed?

**Comment by SuperQ** [comment]
Would you please rebase this in order to fix the generator, we finally fixed it to support the changes in net-snmp.

**Comment by carlosedp** [comment]
Rebased

**Comment by SuperQ** [comment]
You need to revert the changes to snmp.yml.

**Comment by carlosedp** [comment]
Done!

**Review by SuperQ — APPROVED** [review]

**Pull request merged**